src/arrayMethodSort.js

```
@@ -4,8 +4,56 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    const array = this;
+
+    const compareFunctionHelper =
+      typeof compareFunction === 'function'
```

You must throw a `TypeError` if `compareFunction` is provided but is not a function. The current implementation silently falls back to the default comparator, which does not match native sort behavior. Add a check before assigning `compareFunctionHelper` to throw an error if `compareFunction` is not `undefined` and not a function. This is required by checklist item 2.3.